api/src/main/java/com/alibaba/nacos/api/naming/listener/AbstractEventListener.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.api.naming.listener;
+
+import java.util.concurrent.Executor;
+
+/**
+ * Abstract event listener, to support handle event by user custom executor.
+ *
+ * @author horizonzy
+ * @since 1.4.0
+ */
+public abstract class AbstractEventListener implements EventListener {
+    
+    /**
+     * Get executor for execute this receive.
+     *
+     * @return Executor
+     */
+    public Executor getExecutor() {
```

Comment:
Should we abstract this method to EvenListener?

Reply:
Afraid not. If we do this, when user upgrade nacos-client, they need modify code.